imap/imapd.c

```diff
@@ -868,6 +875,16 @@ static void imapd_reset(void)
     /* run delayed commands first before closing anything */
     libcyrus_run_delayed();
 
+    /* log the client behaviors */
+    xsyslog(LOG_NOTICE, "session ended",
+                        "sessionid=<%s> userid=<%s>"
+                        "compress=<%u> notify=<%u>",
+                        session_id(),
+                        imapd_userid ? imapd_userid : "",
+                        client_behavior.did_compress,
+                        client_behavior.did_notify);
+    client_behavior = client_behavior_null;
```

Comment:
Without `client_behaviour_null`, you do this like this:

```suggestion
    memset(&client_behaviour, 0, sizeof(client_behaviour));
```

C99 and later add "compound literals", but we don't really use them in Cyrus (yet).  Using one of those would look something like this, and would be more in the spirit of your original choice (i.e. avoiding a function call):

```suggestion
    client_behavior = (struct client_behaviour_registry){ 0 };
```

Reply:
Thanks, amended!